File: exercises/py101/lesson3_practice_problems/code_review/leap_year.py

# Vlad, your solution is correct.
# It will correctly tell leap or from non-leap years.
# However, it can improve in a few ways.
# First of all, your indentation is not consistent. Try to have 4-space indentation for your python code.
# Secondly, the logic can be improved. The code is going through multiple if statements.
# Going through each if statement, is_a_leap_year is getting reassigned.
# It's worth 

def is_leap_year(year):
    is_a_leap_year = False
    if year % 4 == 0:
        is_a_leap_year = True
    if year % 100 == 0:
        is_a_leap_year = False
    if year % 400 == 0:
        is_a_leap_year = True
    return is_a_leap_year

File: exercises/js210/easy1/leap_year.js

function isLeapYear(year) {
  if (year < 0) {
    console.log(false);
    return false;
  }
  
  if (year % 400 === 0) {
    console.log(true);
    return true;
  } else if (year % 100 === 0) {
    console.log(false);
    return false;
  } else if (year % 4 === 0) {
    console.log(true);
    return true;
  } else {
    console.log(false);
    return false;
  }
}

isLeapYear(2016);      // true
isLeapYear(2015);      // false
isLeapYear(2100);      // false
isLeapYear(2400);      // true
isLeapYear(240000);    // true
isLeapYear(240001);    // false
isLeapYear(2000);      // true
isLeapYear(1900);      // false
isLeapYear(1752);      // true
isLeapYear(1700);      // false
isLeapYear(1);         // false
isLeapYear(100);       // false
isLeapYear(400);       // true